src/components/datepicker/DatePicker.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react';
+import DayPicker, { DayModifiers, DayPickerProps } from 'react-day-picker';
+import Box, { pickBoxProps } from '../box';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import MonthPicker from './MonthPicker';
+import { convertModifiersToClassnames } from './utils';
+import cx from 'classnames';
+import theme from './theme.css';
+import uiUtilities from '@teamleader/ui-utilities';
+import { BoxProps } from '../box/Box';
+import localeUtils from './localeUtils';
+import { GenericComponent } from '../../@types/types';
+
+type Size = 'smallest' | 'small' | 'medium' | 'large';
+
+interface DatePickerProps extends Omit<BoxProps & DayPickerProps, 'size' | 'onChange' | 'modifiers' | 'ref'> {
+  /** If true we give a border to our wrapper. */
+  bordered?: boolean;
+  /** A class name for the DatePicker to give custom styles. */
+  className?: string;
+  /** The modifiers of the DatePicker component. */
+  modifiers?: DayModifiers;
+  /** Callback function that is fired when the date has changed. */
+  onChange?: (day: Date) => void;
+  /** The current selected date. */
+  selectedDate?: Date;
+  /** Size of the DatePicker component. */
+  size?: Size;
+  withMonthPicker?: boolean;
+  showWeekNumbers?: boolean;
+}
+
+const DatePicker: GenericComponent<DatePickerProps> = ({
+  bordered = true,
+  className,
+  modifiers,
+  size = 'medium',
+  withMonthPicker,
+  showWeekNumbers,
+  onChange,
+  ...others
+}) => {
+  const [selectedDate, setSelectedDate] = useState<Date>();
+  const [selectedMonth, setSelectedMonth] = useState<Date>();
+
+  useEffect(() => {
```

Comment:
I think this useEffect can be replaced with a default state on useState 🤔, something like:

`const [selectedDate, setSelectedDate] = useState<Date>(others.selectedDate);`